tools/dev-server/prod.preset.sh

```
@@ -3,110 +3,98 @@
 
-# start the production server
-dotenv \
-  -v NEXT_PUBLIC_GIT_COMMIT_SHA=$(git rev-parse --short HEAD) \
-  -v NEXT_PUBLIC_GIT_TAG=$(git describe --tags --abbrev=0) \
-  -v NEXT_PUBLIC_ICON_SPRITE_HASH="${NEXT_PUBLIC_ICON_SPRITE_HASH}" \
-  "${env_args[@]}" \
-  -- bash -c 'next start -p $NEXT_PUBLIC_APP_PORT' |
+# generate envs.js and start the production server. Both steps belong here rather than in the
+# build: envs.js is read by the browser at runtime, so regenerating it now is what lets a
+# --skip-build run pick up a different --port, and the server itself reads the pro-api flag.
+run_with_envs 'source ./deploy/scripts/export_pro_api_flag.sh && ./deploy/scripts/make_envs_script.sh && next start -p $NEXT_PUBLIC_APP_PORT' |
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)